src/control/server/ioserver/config.go

```diff
@@ -191,12 +191,20 @@ func (c *Config) CmdLineEnv() ([]string, error) {
 	// CaRT and Mercury will now support the new OFI_DOMAIN environment variable so that we can
 	// specify the correct device for each.
 	if strings.Contains(c.Fabric.Provider, "ofi+verbs") {
-		deviceAlias, err := netdetect.GetDeviceAlias(c.Fabric.Interface)
-		if err != nil {
-			return nil, err
+		var skip bool = false
+		for _, envVar := range c.EnvVars {
+			if strings.Contains(envVar, "OFI_DOMAIN") {
```

Comment:
I don't know much about the specifics of this - are we trying to catch a set of environment variables containing "OFI_DOMAIN" or only the exact string?

Reply:
It's looking for OFI_DOMAIN being provided and set to something.  If it has been manually set, then it won't be overridden.

Reply:
I think then you'd want to be more specific - strings.HasPrefix(envVar, "OFI_DOMAIN=") to avoid skipping it for anything in the environment that might just contain "OFI_DOMAIN" as a substring, whether accidentally or on purpose